File: backendApp/src/main/java/com/cinemas/entities/WaterCorn.java
package com.cinemas.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class WaterCorn {
    @Id
    @GeneratedValue
    private Integer id;

    @Column
    private String name;

    @Column
    private String slug;

    @Column
    private Long price;

    @Column
    private String description;

    @Column
    private String image;

    @OneToMany(mappedBy = "waterCorn", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<BookingWaterCorn> bookingWaterCorn;
}

File: fllmbooking_app/lib/data/models/WaterCorn.dart
class WaterCorn {
  int id;
  String name;
  String slug;
  double price;
  String description;
  String image;

  WaterCorn({
    required this.id,
    required this.name,
    required this.slug,
    required this.price,
    required this.description,
    required this.image,
  });

  factory WaterCorn.fromJson(Map<String, dynamic> json) {
    return WaterCorn(
      id: json['id'],
      name: json['name'],
      slug: json['slug'],
      price: json['price'].toDouble(),
      description: json['description'],
      image: json['image'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'slug': slug,
      'price': price,
      'description': description,
      'image': image,
    };
  }

  @override
  String toString() {
    return 'WaterCorn{id: $id, name: $name, slug: $slug, price: $price, description: $description, image: $image}';
  }
}
